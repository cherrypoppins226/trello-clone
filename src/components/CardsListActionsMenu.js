import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Typography from "material-ui/Typography";
import Popover from "material-ui/Popover";
import Divider from "material-ui/Divider";
import { MenuList, MenuItem } from "material-ui/Menu";
import grey from "material-ui/colors/grey";
import * as Labels from "./labels";

const styles = {
  root: {
    "& [role='heading']": {
      paddingTop: 8,
      color: grey[700]
    },
    "& [role='menu']": {
      width: 300,
      "& *": {
        fontWeight: 700
      },
      "& hr": {
        margin: 5,
        marginLeft: 10,
        marginRight: 10
      },
      "& [role='menuitem']": {
        padding: 3,
        paddingLeft: 10,
        transition: "",
        "&:hover": {
          backgroundColor: "rgb(41, 143, 202)",
          "& p": { color: "white" }
        }
      }
    }
  }
};

const CardsListActionsMenu = ({ classes, container, anchor, onClose }) => {
  return (
    <Popover
      className={classes.root}
      aria-describedby={Labels.cardsListActionsMenuDescription.id}
      anchorEl={anchor}
      container={container}
      open={Boolean(anchor)}
      onClose={onClose}
      transitionDuration={0}
      elevation={1}
      anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
      transformOrigin={{ vertical: "top", horizontal: "left" }}
    >
      <Typography role="heading" align="center">
        List Actions
      </Typography>
      <MenuList dense={true} onClick={onClose}>
        {[
          "",
          "Add Card...",
          "Copy List...",
          "Move List...",
          "Watch",
          "",
          "Sort By",
          "",
          "Move All Cards in This List...",
          "Archive All Cards in This List...",
          "",
          "Archive This List"
        ].map((item, idx) => {
          if (!Boolean(item)) return <Divider key={idx} />;
          return (
            <MenuItem key={idx}>
              <Typography> {item} </Typography>
            </MenuItem>
          );
        })}
      </MenuList>
    </Popover>
  );
};

const View = withStyles(styles)(CardsListActionsMenu);

View.propTypes = {
  anchor: PropTypes.object,
  container: PropTypes.object.isRequired,
  onClose: PropTypes.func.isRequired
};

export default View;
