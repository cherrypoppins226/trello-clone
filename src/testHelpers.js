import { render as renderer } from "react-testing-library";
import { createContext } from "react-cosmos-loader";

export const render = (fixture, proxies = []) => {
  const { mount, ...context } = createContext({ renderer, fixture, proxies });

  return (async () => {
    await mount();
    return { ...context.getWrapper(), ...context, rerender: mount };
  })();
};

export const NativeEvents = {
  mouse: {
    click: new MouseEvent("click", { bubbles: true, cancelable: true })
  }
};

export const getByRole = (node, role) => {
  return node.querySelector(`[role="${role}"]`);
};

export const getByAriaLabelled = (node, id) => {
  return node.querySelector(`[aria-labelledby="${id}"]`);
};

export const getByAriaDescribed = (node, id) => {
  return node.querySelector(`[aria-describedby="${id}"]`);
};
