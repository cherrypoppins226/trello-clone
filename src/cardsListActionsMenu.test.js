import React from "react";
import { render } from "./testHelpers.js";
import CardsListActionsMenu from "./cardsListActionsMenu";

it("initial render", () => {
  const container = render(<div />).container.firstElementChild;
  render(
    <CardsListActionsMenu
      anchor={container}
      container={container}
      onClose={jest.fn()}
    />
  );
  expect(container.firstElementChild).not.toBe(null);
  expect(container).toMatchSnapshot();
});
