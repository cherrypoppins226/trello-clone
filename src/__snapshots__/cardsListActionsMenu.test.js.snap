// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div
  style="overflow: hidden; padding-right: 0px;"
>
  <div
    class="MuiModal-root"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiPopover-paper"
      role="document"
      style="opacity: 1; top: 16px; left: 16px;"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter View-listHeader-1"
      >
        List Actions
      </p>
      <ul
        class="MuiList-root MuiList-dense MuiList-padding View-list-2"
        role="menu"
      >
        <hr
          class="MuiDivider-root"
          tabindex="0"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Add Card...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Copy List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Move List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Watch
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Sort By
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Move All Cards in This List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Archive All Cards in This List...
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
          tabindex="-1"
        />
        <li
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
          role="menuitem"
          tabindex="-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             
            Archive This List
             
          </p>
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
